src/sync/types.rs

```diff
@@ -1,22 +1,12 @@
 use serde::{Deserialize, Serialize};
-use std::default::Default;
 
 use crate::{dag, db, util::rlog};
 
 use super::{patch, PullError};
 
-#[derive(Debug, Deserialize, Serialize)]
-#[cfg_attr(test, derive(PartialEq))]
-pub struct BatchPushInfo {
-    #[serde(rename = "httpStatusCode")]
-    pub http_status_code: u16,
-    #[serde(rename = "errorMessage")]
-    pub error_message: String,
-    // TODO BatchPushResponse BatchPushResponse `json:"batchPushResponse"`
-}
-
-#[derive(Clone, Debug, Default, Deserialize, PartialEq, Serialize)]
-pub struct ClientViewInfo {
+#[derive(Serialize, Deserialize)]
+#[cfg_attr(test, derive(Clone, Debug, PartialEq))]
```

Comment:
Not sure if is worth doing the cfg_attr dance? Seems slightly cleaner to have less derives.

Reply:
It doesn't hurt, we should probably be more careful with this kind of thing in general. Suspect as aaron has said there will come a point in time when we find market fit that we do a big push to reduce size.